Binary Search/search_insert_position.js

```diff
@@ -0,0 +1,62 @@
+// https://leetcode.com/problems/search-insert-position/description/
+
+/*
+
+Given a sorted array of distinct integers and a target value, return the index if the target is found. If not, return the index where it would be if it were inserted in order.
+
+You must write an algorithm with O(log n) runtime complexity.
+
+Example 1:
+
+Input: nums = [1,3,5,6], target = 5
+Output: 2
+Example 2:
+
+Input: nums = [1,3,5,6], target = 2
+Output: 1
+Example 3:
+
+Input: nums = [1,3,5,6], target = 7
+Output: 4
+
+*/
+
+function findSearchPosition(nums, target) {
```

Comment:
add an explanation to this problem, update time and space complexity and add more comments to the program 
make sure you [read this doc](https://github.com/akgmage/data-structures-and-algorithms/blob/main/CONTRIBUTING.md)

Reply:
Suggested changes done and pushed, please check again